test/e2e/misc_test.go

```diff
@@ -110,25 +111,30 @@ func (suite *MiscTestSuite) TestFindRoute() {
 	}
 	defer secondContext.Cleanup()
 
+	secondJaegerInstanceName := jaegerInstanceName + "but-even-longer"
 	secondNamespace, err := secondContext.GetNamespace()
-	secondJaegerInstance := getSimplestJaeger(jaegerInstanceName, secondNamespace)
+	secondJaegerInstance := getSimplestJaeger(secondJaegerInstanceName, secondNamespace)
 	err = fw.Client.Create(context.Background(), secondJaegerInstance, cleanupOptions)
 	require.NoError(t, err, "Error deploying example Jaeger")
 	defer undeployJaegerInstance(secondJaegerInstance)
 
 	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout)
 	require.NoError(t, err, "Error waiting for deployment")
 
-	err = e2eutil.WaitForDeployment(t, fw.KubeClient, secondNamespace, jaegerInstanceName, 1, retryInterval, timeout)
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, secondNamespace, secondJaegerInstanceName, 1, retryInterval, timeout)
 	require.NoError(t, err, "Error waiting for deployment")
 
 	// Make sure findRoute returns the correct routes for each namespace
 	route := findRoute(t, fw, jaegerInstanceName, namespace)
 	require.Equal(t, namespace, route.Namespace)
+	truncatedInstanceName := util.DNSName(util.Truncate(jaegerInstanceName, 62-len(namespace)))
+	require.True(t, strings.HasPrefix(route.Spec.Host, truncatedInstanceName))
 	require.True(t, strings.Contains(route.Spec.Host, namespace))
 
-	secondRoute := findRoute(t, fw, jaegerInstanceName, secondNamespace)
+	secondRoute := findRoute(t, fw, secondJaegerInstanceName, secondNamespace)
 	require.Equal(t, secondNamespace, secondRoute.Namespace)
+	secondTruncatedInstanceName := util.DNSName(util.Truncate(secondJaegerInstanceName, 62-len(namespace)))
```

Comment:
Shouldn't this be `len(secondNamespace)`?

Reply:
Would it be possible to add a "not equals" check to show `secondTruncatedInstanceName` is not same as `secondJaegerInstanceName`?